eng/common/pipelines/templates/jobs/docindex.yml

```diff
@@ -1,7 +1,8 @@
 jobs:
   - job: CreateDocIndex
     pool:
-      name: azsdk-pool-mms-win-2022-general
+      name: azsdk-pool
```

Comment:
Should we use the variables here?

Is there any reason you know of this cannot be linux?

Reply:
Variables: There is no `eng/common/pipelines/templates/variables/image.yml` today. I could introduce one. 

Linux: It could probably run on Linux. Will need to test each docindex pipeline... can do that after merging the other PRs.

Reply:
I'm suggesting you assume those variables are present and any consumers of this will have to set them up. We do something similar here https://github.com/Azure/azure-sdk-tools/blob/main/eng/common/pipelines/templates/jobs/prepare-pipelines.yml#L16